My Series List/My Series List/View/SearchView/SearchPresenter.swift

```diff
@@ -5,32 +5,84 @@
+    func listEmpty()
 }
 
 class SearchPresenter {
     
     var searchView: SearchView?
     let seriesRepository = SeriesRepository()
+    var series: [Api.Series] = []
+    var seriesSorted: [Api.Series] = []
     
     func searchSeries(search: String) {
-        seriesRepository.searchSeries(search: search, completionHandler: { series in
-            switch series {
+        let replacedText = search.replacingOccurrences(of: " ", with: "+")
+        seriesRepository.searchSeries(search: replacedText, completionHandler: { results in
+            switch results {
             case .failure(let error):
-                print(error.localizedDescription)
-                self.searchView?.listEmpty(error)
-            case .success(let series):
+                self.searchView?.listError(error)
+            case .success(let seriesApi):
+                self.series = seriesApi.results
+                if self.series.isEmpty {
+                    self.searchView?.listEmpty()
+                }
                 self.searchView?.startLoading()
-                self.searchView?.listSeries(series)
+                DispatchQueue.main.async {
+                    self.checkRepeated(seriesArray: self.series)
+                    self.seriesSorted = self.series.sorted(by: { $0.title! < $1.title! })
+                    self.searchView?.listSeries(self.seriesSorted)
+                }
                 self.searchView?.finishLoading()
             }
          })
     }
+    
+    func changeListCheckMark(_ checkMark: UIButton, indexPath: IndexPath) {
+        if seriesSorted[indexPath.row].isSaved == false {
+            checkMark.setImage(UIImage(systemName: "checkmark.rectangle.fill"), for: .normal)
+            seriesRepository.insertSeries(series: seriesSorted[indexPath.row])
+            seriesSorted[indexPath.row].isSaved = true
+        } else {
+            checkMark.setImage(UIImage(systemName: "checkmark.rectangle"), for: .normal)
+            seriesSorted[indexPath.row].isSaved = false
+            seriesRepository.getSeries { results in
+                switch results {
+                case .failure(_):
+                    break
+                case .success(let seriesCoreData):
+                    for series in seriesCoreData {
+                        if series.id == self.seriesSorted[indexPath.row].id {
+                            self.seriesRepository.deleteSeries(series: series) { results in
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+    
+    func checkRepeated(seriesArray: [Api.Series]) {
+        seriesRepository.getSeries { results in
+            switch results {
+            case .failure(_):
+                break
+            case .success(let seriesCoreData):
+                for seriesA in seriesCoreData {
+                    for seriesB in seriesArray {
+                        if seriesA.id == seriesB.id! {
```

Comment:
You should use break in order to skip checking all values.
In my understanding you might even want to switch the order of the for cycles in order to iterate each item from seriesArray once and for each one checking values on repository